lib/gooddata/lcm/actions/collect_client_projects.rb

```diff
@@ -33,7 +33,9 @@ def call(params)
           client = params.gdc_gd_client
 
           domain = client.domain(params.organization) || fail("Invalid domain name specified - #{params.organization}")
-          all_segments = domain.segments
+
+          data_product = params.data_product
```

Comment:
Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)